pageserver/src/tenant/layer_map.rs

```diff
@@ -63,6 +64,22 @@ pub use historic_layer_coverage::Replacement;
 
 use super::storage_layer::range_eq;
 
+struct LayerRef<L: ?Sized>(Arc<L>);
+
+impl<L: ?Sized> PartialEq for LayerRef<L> {
+    fn eq(&self, other: &LayerRef<L>) -> bool {
+        Arc::ptr_eq(&self.0, &other.0)
```

Comment:
Clippy probably doesn't anything because now there's no `dyn Layer` info ever in this context -- note that there's no bound on `L` except `?Sized`. Lack of bound means that when looking at this function, clippy will never have information on the actual usage with `dyn Layer` because it's already erased.

Though I am unsure why is this LayerRef needed all. It's just an arc.

Reply:
This was solved with #3558 and #3573.

Reply:
I have already integrated this changes in this PR